src/snowflake/cli/plugins/nativeapp/codegen/snowpark/callback_source.py.jinja

```diff
@@ -12,107 +14,144 @@ except ModuleNotFoundError as exc:
     )
     sys.exit(1)
 
-orig_globals = globals().copy()
-
-found_correct_version = hasattr(
+__snowflake_internal_found_correct_version = hasattr(
     snowflake.snowpark.context, "_is_execution_environment_sandboxed_for_client"
 ) and hasattr(snowflake.snowpark.context, "_should_continue_registration")
 
-if not found_correct_version:
+if not __snowflake_internal_found_correct_version:
     print(
         "Did not find the minimum required version for snowflake-snowpark-python package. Please upgrade to v1.15.0 or higher.",
         file=sys.stderr,
     )
     sys.exit(1)
 
-__snowflake_cli_native_app_internal_callback_return_list: List[Any] = []
+__snowflake_global_collected_extension_fn_json = []
+
+def __snowflake_internal_create_extension_fn_registration_callback():
+    def __snowflake_internal_try_extract_lineno(extension_fn):
+        try:
+            return inspect.getsourcelines(extension_fn)[1]
```

Comment:
[1] to omit the decorator?
Also, would it make sense to return `inspect.getsourcelines(extension_fn)[1:]` instead, especially for de-annotation work, instead of only `def` of the function?

Reply:
AFAIK getsourcelines only returns the first line number of the function declaration. It won't help with deannotation, sadly.

Reply:
Oh I see what you're asking. `getsourcelines` is peculiar, it returns a pair of `(source code of the function, first lineno)`. So I discard the source code and keep the line number.